session.go

```diff
@@ -1434,10 +1467,45 @@ type Iter struct {
 	next    *nextIter
 	host    *HostInfo
 
+	metrics *queryMetrics
+
 	framer *framer
 	closed int32
 }
 
+func NewIter() *Iter {
```

Comment:
Is there a better way to handle all distinct ways `Iter` is constructed?

Reply:
this is probably fine but I'd say not to expose these functions because it's not really expected for users to be creating `Iter` objects anyway